sky/backends/cloud_vm_ray_backend.py

```diff
@@ -1732,11 +1753,18 @@ def provision_with_retries(
             logger.warning(f'\n{style.BRIGHT}Provision failed for {num_nodes}x '
                            f'{to_provision} in {region_or_zone_str}. '
                            f'Trying other locations (if any).{style.RESET_ALL}')
-            if not cluster_exists:
+            if prev_cluster_status is None:
                 # Add failed resources to the blocklist, only when it
                 # is in fallback mode.
                 self._blocked_resources.add(to_provision)
             else:
+                # If reached here, it means that the existing cluster should
+                # have been INIT and should have been terminated, as other cases
+                # will not trigger the failover due to `no_failover` flag.
```

Comment:
```suggestion
                # If we reach here, it means that the existing cluster must have a previous status of INIT, because other statuses (UP, STOPPED) will not trigger the failover due to `no_failover` flag; see _yield_zones(). Also, the cluster should have been terminated by _retry_zones().
```

Reply:
(not for this PR) This reminds of this:

```

            # If cluster was previously UP or STOPPED, stop it; otherwise
            # terminate.
            # FIXME(zongheng): terminating a potentially live cluster is
            # scary. Say: users have an existing cluster that got into INIT, do
            # sky launch, somehow failed, then we may be terminating it here.
```
which seems pretty scary still?

Reply:
Fixed. Thanks!

Yes! This is still a big problem. Any clusters that are unexpectedly set to INIT mode can be accidentally terminated (e.g., ray up fails not because the resources unavailability) if we do `sky launch -c existing-cluster`.